templates/editConn.js

```diff
@@ -1,98 +1,118 @@
-const vscode = acquireVsCodeApi();
-let err = [];
-save();
-testConn();
-
-function save() {
-  document.getElementById("save").addEventListener("click", () => {
-    let data = getData();
-    if (err.length > 0) {
-      return {};
-    }
-    vscode.postMessage({
-      command: "save",
-      connID: data.connID,
-      connName: data.connName,
-      connHost: data.connHost,
-      connToken: data.connToken,
-      connOrg: data.connOrg,
-      connDefault: data.connDefault
-    });
-  });
-}
+class Actions {
+  constructor(hasErr, vscode) {
+    this.hasErr = hasErr;
+    this.vscode = vscode;
+  }
 
-function testConn() {
-  document.getElementById("testConn").addEventListener("click", () => {
-    let data = getData();
-    if (err.length > 0) {
-      return {};
-    }
-    vscode.postMessage({
-      command: "testConn",
-      connName: data.connName,
-      connHost: data.connHost,
-      connToken: data.connToken,
-      connOrg: data.connOrg
+  save() {
+    let self = this;
+    document.getElementById("save").addEventListener("click", () => {
+      let data = self.getData();
+      if (self.hasErr) {
+        return {};
+      }
+      self.vscode.postMessage({
+        command: "save",
+        connID: data.connID,
+        connName: data.connName,
+        connHost: data.connHost,
+        connToken: data.connToken,
+        connOrg: data.connOrg,
+        connDefault: data.connDefault
+      });
     });
-  });
-}
+  }
 
-function getData() {
-  err = [];
-  let connID = document.getElementById("connID").value;
-  let isDefault = document.getElementById("connDefault").checked;
-  let connName = validTextInput("connName", "Name is empty for the connection");
-  let connHost = validTextInput(
-    "connHost",
-    "Host URI is empty for the connection"
-  );
-  let connToken = validTextInput(
-    "connToken",
-    "Token is empty for the connection",
-    "^[-A-Za-z0-9+=]{1,50}|=[^=]|={3,}$",
-    "Token should be a valid base64 encoded string"
-  );
-  let connOrg = validTextInput("connOrg", "Org is empty for the connection");
-  let errMsgs = document.getElementById("errMsgs");
-  if (err.length > 0) {
-    errMsgs.innerHTML = "";
-    errMsgs.classList.remove("hidden");
-    err.forEach(ele => {
-      var pNode = document.createElement("p"); // Create a <li> node
-      pNode.appendChild(document.createTextNode(ele));
-      errMsgs.appendChild(pNode);
+  testConn() {
+    let self = this;
+    document.getElementById("testConn").addEventListener("click", () => {
+      let data = self.getData();
+      if (self.hasErr) {
+        return {};
+      }
+      self.vscode.postMessage({
```

Comment:
this is a bound function, you can just use this here and remove the self variable.